service/worker/replicator/replicator.go

```diff
@@ -243,9 +243,13 @@ func isRetryableError(err error) bool {
 	switch err.(type) {
 	case *serviceerror.Internal,
 		*serviceerrors.ShardOwnershipLost,
-		*serviceerror.DeadlineExceeded,
 		*serviceerror.Unavailable:
 		return true
 	}
+
+	if common.IsContextTimeoutErr(err) {
+		return true
```

Comment:
this should be false?

since this retry function is used within client wrapper, so once timeout, always timeout until context is changed?

Reply:
I don't know. This is just parity with what it was. There is a mess with all those `IsRetry` and `IsTransient` funcs. They need to be properly reviewed.

Reply:
This one is actually used by kafka processor only. So it doesn't matter at all.

Reply:
it is actually used here: https://github.com/temporalio/temporal/pull/890/files/7a9ba91cc0d1bb9adc008ab6b8b25ba4c45a9adf#diff-b20c04dd2e72666b9c2dfe62e823a3f498e72b1a04d0d0e4edeebd3e0a29ac15R171

you can keep this as is, i will deal with it once i start on kafka deprecation for ndc